src/main/java/com/diablo3CharViewer/CharacterViewerManager.java

```diff
@@ -4,33 +4,36 @@
 import com.diablo3CharViewer.api_handlers.ItemHandlerApi;
 import com.diablo3CharViewer.json_mappers.AccountMapper;
 import com.diablo3CharViewer.json_mappers.HeroMapper;
+import com.diablo3CharViewer.json_mappers.ItemMapper;
 import com.diablo3CharViewer.token.FetchToken;
 
 import java.util.Scanner;
 
 public class CharacterViewerManager {
 
-    public void showProfile(Scanner scanner, AccountMapper accountMapper, FetchToken fetchToken) {
+    public String profileDataModelProvider(Scanner scanner, AccountMapper accountMapper, FetchToken fetchToken) {
 
         String battleTag = battleTagProvider(scanner);
-        if(isBattleTagCorrect(battleTag)) {
-            System.out.println("Zostan na chwile i poczytaj:\n" + accountMapper.fetchAccountToDataModel(battleTag, fetchToken) + '\n');
+        if (isBattleTagCorrect(battleTag)) {
+            return "Zostan na chwile i poczytaj:\n" + accountMapper.fetchAccountToDataModel(battleTag, fetchToken) + '\n';
         }
+        return "";
     }
 
-    public void showHero(Scanner scanner, HeroMapper heroMapper, FetchToken fetchToken) {
+    public String heroDataModelProvider(Scanner scanner, HeroMapper heroMapper, FetchToken fetchToken) {
 
         String battleTag = battleTagProvider(scanner);
         if (isBattleTagCorrect(battleTag)) {
             String heroId = heroIdProvider(scanner);
             if (isHeroIDCorrect(heroId)) {
-                System.out.println("Zostan na chwile i poczytaj:\n" + heroMapper.fetchHeroToDataModel(battleTag, heroId, fetchToken) + '\n');
+                return "Zostan na chwile i poczytaj:\n" + heroMapper.fetchHeroToDataModel(battleTag, heroId, fetchToken) + '\n';
             }
         }
+        return "";
```

Comment:
Why empty String returned?

Reply:
Now notifications flow in this function is changed and warnings are returen as strings in if-else statements.